pkg/proxy/apis/config/v1alpha1/defaults.go

```diff
@@ -37,16 +39,49 @@ func SetDefaults_KubeProxyConfiguration(obj *kubeproxyconfigv1alpha1.KubeProxyCo
 	if len(obj.BindAddress) == 0 {
 		obj.BindAddress = "0.0.0.0"
 	}
-	if obj.HealthzBindAddress == "" {
-		obj.HealthzBindAddress = fmt.Sprintf("0.0.0.0:%v", ports.ProxyHealthzPort)
-	} else if !strings.Contains(obj.HealthzBindAddress, ":") {
-		obj.HealthzBindAddress += fmt.Sprintf(":%v", ports.ProxyHealthzPort)
+	// ipv6
```

Comment:
Humm..The ipv6 and ipv4 block looks fairly complicated. It also seems like we are duplicating the logic to handle `HealthzBindAddress` and `MetricsBindAddress`. We should extract out the common part and reuse if possible.

Putting such logic into a func would also help unit testing.

Reply:
It just considers ipv4 format originally.
Now we have ipv6 format have to check.
As the following link that can make sure ``BindAddress`` and ``HealthzBindAddress`` format are ipv4 or ipv6.
But there are some unit tests failed, I'm not sure is that some test cases I haven't thought
https://github.com/kubernetes/kubernetes/pull/76320/files#diff-f1bfd0a64c0e2e6163dfbf23b47abe93R44

Reply:
I haven't looked closely, just adding a quick note that some tests are failing because the bazel file is not updated (https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/76320/pull-kubernetes-verify/1116634758399397889). You might need to run `./hack/update-bazel.sh`.

Reply:
Good. It fixes almost.
Now only ``pull-kubernetes-e2e-gce`` is left.
Thanks for the help.
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/76320/pull-kubernetes-e2e-gce/1116758947001274377/

Reply:
Took another look, it still seems to me this is more complicated than it needs to be. I gave it a try on https://github.com/MrHohn/kubernetes/commit/c37f7062429a8cecfa3d5bddbe85c8daf55cf3a6. Do you mind taking a look if that works for you? Thanks.

Reply:
LGTM
Thanks

Reply:
I make a little change. It should give ipv6 format when bindAddress is ipv6, that is one thing the PR have to refactor